Source/HTTP/httpcall.cpp

```diff
@@ -353,6 +368,32 @@ void HC_CALL::PerformSingleRequestComplete(XAsyncBlock* async)
         }
     }
 
+    // Decompress Response Bytes 
+    if (Compression::Available() && call->compressedResponse == true)
+    {
+        http_internal_vector<uint8_t> uncompressedResponseBodyBuffer;
+
+        Compression::DecompressFromGzip(
+            call->responseBodyBytes.data(),
+            call->responseBodyBytes.size(),
+            uncompressedResponseBodyBuffer);
+
+        call->responseBodyBytes.resize(uncompressedResponseBodyBuffer.size());
+        call->responseBodyBytes = std::move(uncompressedResponseBodyBuffer);
+    }
+
+    // Check if we 'reset' the custom response write callback before decompressing the response
+    HCHttpCallResponseBodyWriteFunction temporaryWriteFunction = call->clientResponseBodyWriteFunction;
+
+    // call->temporaryResponseBodyWriteFunction should remain HC_CALL::ResponseBodyWrite if we did not 'reset' the custom response write callback
```

Comment:
Update comment.  I also think this could be safely moved inside the above if statement, though I don't think it makes a big difference